catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -0,0 +1,233 @@
+import org.codice.ddf.catalog.ui.query.cql.CqlQueryResponse;
+import org.codice.ddf.catalog.ui.query.cql.CqlRequest;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.eclipse.jetty.http.HttpStatus;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+import spark.utils.IOUtils;
+
+public class CqlTransformHandler implements Route {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CqlTransformHandler.class);
+
+  private List<ServiceReference> queryResponseTransformers;
+
+  private BundleContext bundleContext;
+
+  private static final String GZIP = "gzip";
+
+  private ObjectMapper mapper =
+      new ObjectMapperImpl(
+          new JsonParserFactory().usePropertyOnly(),
+          new JsonSerializerFactory()
+              .includeEmpty()
+              .includeNulls()
+              .includeDefaultValues()
+              .setJsonFormatForDates(false)
+              .useAnnotations());
+
+  private EndpointUtil util;
+
+  public CqlTransformHandler(
+      List<ServiceReference> queryResponseTransformers,
+      BundleContext bundleContext,
+      EndpointUtil endpointUtil) {
+    this.queryResponseTransformers = queryResponseTransformers;
+    this.bundleContext = bundleContext;
+    this.util = endpointUtil;
+  }
+
+  public class Arguments {
+    private Map<String, Object> arguments;
+
+    public Arguments() {
+      this.arguments = Collections.emptyMap();
+    }
+
+    public void setArguments(Map<String, Object> arguments) {
+      this.arguments = arguments;
+    }
+
+    public Map<String, Object> getArguments() {
+      return this.arguments;
+    }
+
+    public Map<String, Serializable> getSerializableArguments() {
+      Map<String, Serializable> serializableMap =
+          this.getArguments()
+              .entrySet()
+              .stream()
+              .filter(entry -> entry.getValue() instanceof Serializable)
+              .collect(Collectors.toMap(Map.Entry::getKey, e -> (Serializable) e.getValue()));
+      return serializableMap;
+    }
+  }
+
+  @Override
+  public Object handle(Request request, Response response) throws Exception {
+    String transformerId = request.params(":transformerId");
+    String body = util.safeGetBody(request);
+    CqlRequest cqlRequest;
+
+    try {
+      cqlRequest = mapper.readValue(body, CqlRequest.class);
+      if (cqlRequest == null) {
+        LOGGER.debug("Cql request parsed from body evaluated to  null.");
+        throw new IllegalArgumentException("Cql request is null.");
+      }
+    } catch (Exception e) {
+      LOGGER.debug("Error fetching cql request");
+      response.status(HttpStatus.BAD_REQUEST_400);
+      return ImmutableMap.of("message", "Bad request");
+    }
+
+    Arguments arguments = mapper.readValue(body, Arguments.class);
```

Comment:
i thought of that but i believe it would be much trickier.
certain QueryResponseTransformers have different types of arguments that can be passed to them. so to leave it arbitrary I left it to be sent as json, which is what the csv transformer similarly does.

overall I'm just not sure how it would work as url query params. 
it could be difficult to separate values if some are meant to be comma separated and others aren't. 
and the parameters can't be verbatim for instance ?hiddenvalues=x,y,z&columnorder=a,b,c
some transformers may need something different from `hidden-values` and `column-ordering`, and since transformer arguments are already read as a map it makes sense to have the json `arguments: ` element be read in the same